### PR TITLE
Added tf_logging for deprecated 'experimental_run_v2' (Tf 1.x)

### PR DESCRIPTION
**Pull request by abhilash1910**
To resolve PR #49187 , by providing logging messages for  deprecated 'experimental_run_v2'  in Tf 1.x

### CODE REVIEW

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]
I don't see how this resolves the issue

**Comment by abhilash1910** [comment]
Hi @mihaimaruseac ,
could you please specify the changes required in this case?  This PR was for adding a logging message regarding the tf version specification.

**Comment by mihaimaruseac** [comment]
But why add the logging messages? We don't add these messages to the other deprecated APIs

**Comment by mihaimaruseac** [comment]
See also https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/util/deprecation.py

**Comment by abhilash1910** [comment]
Hi, 
Thanks for the link, I think the 'tf_logging' can be done away with, and those can be replaced with the ```@depricated``` decorator instead?
For instance:
```python
@doc_controls.do_not_doc_inheritable  # DEPRECATED, moving to `extended` 
@deprecated(None,'use extended.colocate_vars_with() instead.')
 def colocate_vars_with(self, colocate_with_variable):   
      """DEPRECATED: use extended.colocate_vars_with() instead."""    
      return self._extended.colocate_vars_with(colocate_with_variable)
```
Thanks

File: tensorflow/python/distribute/distribute_lib.py [hunk 1]
```diff
@@ -937,13 +937,15 @@ def scope(self):
...
   @doc_controls.do_not_doc_inheritable  # DEPRECATED, moving to `extended`
+  @deprecated(None,'use extended.colocate_vars_with() instead.')
   def colocate_vars_with(self, colocate_with_variable):
     """DEPRECATED: use extended.colocate_vars_with() instead."""
     return self._extended.colocate_vars_with(colocate_with_variable)
 
   @doc_controls.do_not_generate_docs  # DEPRECATED: TF 1.x only
   def make_dataset_iterator(self, dataset):
     """DEPRECATED TF 1.x ONLY."""
+
```
Comment (mihaimaruseac) [inline tensorflow/python/distribute/distribute_lib.py:948]:
```suggestion
```

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]
You should add it to all deprecated methods

File: tensorflow/python/distribute/distribute_lib.py (hunk 1 above)
Comment (abhilash1910) [reply tensorflow/python/distribute/distribute_lib.py]: Resolved. And added deprecation decorators where-ever specified.

**Comment by abhilash1910** [comment]
I have added the deprecated decorators ,please let me know if any changes are required

**Comment by mihaimaruseac** [comment]
Adding Ran to review the distrstrat changes

File: tensorflow/python/distribute/distribute_lib.py [hunk 2]
```diff
@@ -1842,7 +1850,7 @@ class StrategyV1(StrategyBase):
   TensorFlow's partitioned variables (where a single variable is split across
   multiple devices) at this time.
   """
-
+  @deprecated(None,'This method is not available in TF 2.x.')
```
Comment (crccw) [inline tensorflow/python/distribute/distribute_lib.py:1853]: This is the TF1 Strategy base object. I don't think we would like to have deprecation warnings for TF1 users.

Comment (abhilash1910) [reply tensorflow/python/distribute/distribute_lib.py]: Reverted the changes for TF1 Strategy based objects.

**Comment by abhilash1910** [comment]
Reverted the decorator changes for TF1.x Strategy objects , please let me know if any changes are required.

**Comment by abhilash1910** [comment]
Hi @crccw  @mihaimaruseac , I see that after merging the master in my remote, google butler removed the ready to pull tag. Can the tag be re-assigned?

File: tensorflow/python/distribute/distribute_lib.py [hunk 3]
```diff
@@ -2866,6 +2873,7 @@ def read_var(self, v):
     """
     raise NotImplementedError("must be implemented in descendants")
 
+
```
Comment (mihaimaruseac) [inline tensorflow/python/distribute/distribute_lib.py:2876]: Please don't insert additional whitespace

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: tensorflow/python/distribute/distribute_lib.py (hunk 3 above)
Comment (abhilash1910) [reply tensorflow/python/distribute/distribute_lib.py]: Removed additional whitespace.

**Comment by abhilash1910** [comment]
Resolved spaces, let me know if any other changes required.

File: tensorflow/python/distribute/distribute_lib.py [hunk 4]
```diff
@@ -959,6 +960,7 @@ def make_input_fn_iterator(self,
           input_fn, replication_mode=replication_mode)
 
   @doc_controls.do_not_generate_docs  # DEPRECATED: TF 1.x only
+  @deprecated(None,'use mirrored_strategy.run() instead')
```
Comment (crccw) [inline tensorflow/python/distribute/distribute_lib.py:963]: use run() instead.

**Review by crccw — CHANGES_REQUESTED** [review]

File: tensorflow/python/distribute/distribute_lib.py (hunk 4 above)
Comment (abhilash1910) [reply tensorflow/python/distribute/distribute_lib.py]: Made the required change at line 963. Let me know if any other changes are required.

File: tensorflow/python/distribute/distribute_lib.py [hunk 5]
```diff
@@ -959,6 +960,7 @@ def make_input_fn_iterator(self,
           input_fn, replication_mode=replication_mode)
 
   @doc_controls.do_not_generate_docs  # DEPRECATED: TF 1.x only
+  @deprecated(None,'use run() instead')
```
Comment (crccw) [inline tensorflow/python/distribute/distribute_lib.py:963]: whitespace at the end of the line

File: tensorflow/python/distribute/distribute_lib.py [hunk 6]
```diff
@@ -1869,6 +1872,7 @@ def make_dataset_iterator(self, dataset):
     """
     return self._extended._make_dataset_iterator(dataset)  # pylint: disable=protected-access
 
+
```
Comment (crccw) [inline tensorflow/python/distribute/distribute_lib.py:1875]: whitespace

File: tensorflow/python/distribute/distribute_lib.py [hunk 7]
```diff
@@ -1982,6 +1987,7 @@ def reduce(self, reduce_op, value, axis=None):
 
   reduce.__doc__ = StrategyBase.reduce.__doc__
 
+
```
Comment (crccw) [inline tensorflow/python/distribute/distribute_lib.py:1990]: whitespace

File: tensorflow/python/distribute/distribute_lib.py [hunk 8]
```diff
@@ -2895,7 +2902,7 @@ def update_non_slot(
 
   def _update_non_slot(self, colocate_with, fn, args, kwargs, group):
     raise NotImplementedError("must be implemented in descendants")
-
+
```
Comment (crccw) [inline tensorflow/python/distribute/distribute_lib.py:2905]: whitespace

File: tensorflow/python/distribute/distribute_lib.py [hunk 9]
```diff
@@ -1842,7 +1846,6 @@ class StrategyV1(StrategyBase):
   TensorFlow's partitioned variables (where a single variable is split across
   multiple devices) at this time.
   """
-
```
Comment (crccw) [inline tensorflow/python/distribute/distribute_lib.py:1845]: Please add back the blank line.

**Review by crccw — CHANGES_REQUESTED** [review]

File: tensorflow/python/distribute/distribute_lib.py (hunk 5 above)
Comment (abhilash1910) [reply tensorflow/python/distribute/distribute_lib.py]: Have removed the whitespace

File: tensorflow/python/distribute/distribute_lib.py (hunk 6 above)
Comment (abhilash1910) [reply tensorflow/python/distribute/distribute_lib.py]: Whitespace is removed

File: tensorflow/python/distribute/distribute_lib.py (hunk 7 above)
Comment (abhilash1910) [reply tensorflow/python/distribute/distribute_lib.py]: Whitespace is removed

File: tensorflow/python/distribute/distribute_lib.py (hunk 8 above)
Comment (abhilash1910) [reply tensorflow/python/distribute/distribute_lib.py]: This whitespace is a single at 2905.

File: tensorflow/python/distribute/distribute_lib.py (hunk 9 above)
Comment (abhilash1910) [reply tensorflow/python/distribute/distribute_lib.py:1845]: Added back the blank line

Comment (mihaimaruseac) [reply tensorflow/python/distribute/distribute_lib.py:1845]: You also added trailing whitespace here

File: tensorflow/python/distribute/distribute_lib.py [hunk 10]
```diff
@@ -2895,7 +2901,7 @@ def update_non_slot(
 
   def _update_non_slot(self, colocate_with, fn, args, kwargs, group):
     raise NotImplementedError("must be implemented in descendants")
-
+
```
Comment (mihaimaruseac) [inline tensorflow/python/distribute/distribute_lib.py:2904]: Please don't add trailing whitespace at the end of lines.
```suggestion

```

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: tensorflow/python/distribute/distribute_lib.py (hunk 10 above)
Comment (abhilash1910) [reply tensorflow/python/distribute/distribute_lib.py]: Removed the trailing whitespace

**Review by mihaimaruseac — APPROVED** [review]

**Pull request merged**